internal/component/otelcol/exporter/debug/debug.go

```diff
@@ -0,0 +1,97 @@
+package debug
+
+import (
+	"fmt"
+
+	"github.com/grafana/alloy/internal/component"
+	"github.com/grafana/alloy/internal/component/otelcol"
+	otelcolCfg "github.com/grafana/alloy/internal/component/otelcol/config"
+	"github.com/grafana/alloy/internal/component/otelcol/exporter"
+	"github.com/grafana/alloy/internal/featuregate"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configtelemetry"
+	"go.opentelemetry.io/collector/exporter/debugexporter"
+	otelextension "go.opentelemetry.io/collector/extension"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "otelcol.exporter.debug",
+		Stability: featuregate.StabilityExperimental,
+		Args:      Arguments{},
+		Exports:   otelcol.ConsumerExports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := debugexporter.NewFactory()
+			return exporter.New(opts, fact, args.(Arguments), exporter.TypeAll)
+		},
+	})
+}
+
+type Arguments struct {
+	Verbosity          string `alloy:"verbosity,attr,optional"`
+	SamplingInitial    int    `alloy:"sampling_initial,attr,optional"`
+	SamplingThereafter int    `alloy:"sampling_thereafter,attr,optional"`
+
+	// DebugMetrics configures component internal metrics. Optional.
+	DebugMetrics otelcolCfg.DebugMetricsArguments `alloy:"debug_metrics,block,optional"`
+}
+
+func (args Arguments) convertVerbosity() (configtelemetry.Level, error) {
+	var verbosity configtelemetry.Level
+	switch args.Verbosity {
+	case "basic":
+		verbosity = configtelemetry.LevelBasic
+	case "normal":
+		verbosity = configtelemetry.LevelNormal
+	case "detailed":
+		verbosity = configtelemetry.LevelDetailed
+	default:
+		// Invalid verbosity
+		// debugexporter only supports basic, normal and detailed levels
+		return verbosity, fmt.Errorf("invalid verbosity %q", args.Verbosity)
+	}
+
+	return verbosity, nil
+}
+
+var _ exporter.Arguments = Arguments{}
+
+// SetToDefault implements river.Defaulter.
+func (args *Arguments) SetToDefault() {
+	*args = Arguments{
+		Verbosity:          "normal",
```

Comment:
upstream's default is set to basic, even if normal and basic are the same, we should maybe align on the exact same values: https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/debugexporter/README.md